lib/checkstl.cpp

```diff
@@ -2610,10 +2610,24 @@ void CheckStl::useStlAlgorithm()
                 continue;
             const Token *bodyTok = tok->next()->link()->next();
             const Token *splitTok = tok->next()->astOperand2();
-            if (!Token::simpleMatch(splitTok, ":"))
-                continue;
-            const Token *loopVar = splitTok->previous();
-            if (!Token::Match(loopVar, "%var%"))
+            const Token* loopVar{};
+            if (Token::simpleMatch(splitTok, ":")) {
+                loopVar = splitTok->previous();
+                if (!Token::Match(loopVar, "%var%"))
+                    continue;
+            }
+            else if (Token::simpleMatch(splitTok, ";")) { // iterator-based loop
+                loopVar = splitTok->next();
+                if (!Token::Match(loopVar, "%var%") || !loopVar->valueType() || loopVar->valueType()->type != ValueType::Type::ITERATOR)
+                    continue;
+                const Token* initAssign = splitTok->astOperand1();
+                if (!Token::simpleMatch(initAssign, "=") || !Token::Match(initAssign->astOperand1(), "%var%", loopVar->varId()))
```

Comment:
`%var%` => `%varid%`

Reply:
We really should add an internal check for that - see https://trac.cppcheck.net/ticket/10987.